File: tft-service/models/summoner.js
const Sequelize = require('sequelize');
const { search } = require('../app');

const sequelize = new Sequelize('tftSQL', 'root', 'zxcdsa1998', {
    host: 'localhost',
    dialect: 'mysql'
  });

const Summoner = sequelize.define('summoners', {
    /*
        profileIconId	int	ID of the summoner icon associated with the summoner.
        name	string	Summoner name.
        puuid	string	Encrypted PUUID. Exact length of 78 characters.
        summonerLevel	long	Summoner level associated with the summoner.
        revisionDate	long	Date summoner was last modified specified as epoch milliseconds. The following events will update this timestamp: profile icon change, playing the tutorial or advanced tutorial, finishing a game, summoner name change
        id	string	Encrypted summoner ID. Max length 63 characters.
        accountId	string	Encrypted account ID. Max length 56 characters.
    */
    // attributes
    profileIconId: {
      type: Sequelize.INTEGER,
    },
    name: {
      type: Sequelize.STRING
    },
    puuid: {
        type: Sequelize.STRING,
        allowNull: false,
        primaryKey: true,
    },
    summonerLevel: {
        type: Sequelize.BIGINT
    },
    revisionDate: {
        type: Sequelize.BIGINT,
    },
    id: {
        type: Sequelize.STRING,
        allowNull: false,
    },
    accountId: {
        type: Sequelize.STRING,
        allowNull: false,
    },
    tier: {
      type: Sequelize.STRING,
      allowNull: true,    
    },
    rank: {
      type: Sequelize.STRING,
      allowNull: true,
    },
    wins: {
      type: Sequelize.INTEGER,
      allowNull: true,
    },
    losses: {
      type: Sequelize.INTEGER,
      allowNull: true,
    },
    leaguePoints: {
      type: Sequelize.INTEGER,
      allowNull: true
    }
  }, {
    // options
    sequelize,
    modelName: 'summoners'
  });

  module.exports = Summoner;

File: tft_app/lib/models/summoner.dart
class Summoner {
  final int profileIconId;
  final String name;
  final String puuid;
  final int summonerLevel;
  final int revisionDate;
  final String id;
  final String accountId;
  final String tier;
  final String rank;
  final int wins;
  final int losses;
  final int leaguePoints;
  Summoner(
      {this.profileIconId,
      this.name,
      this.puuid,
      this.summonerLevel,
      this.revisionDate,
      this.id,
      this.accountId,
      this.tier,
      this.rank,
      this.wins,
      this.losses,
      this.leaguePoints});

  factory Summoner.fromJson(Map<String, dynamic> json) {
    return Summoner(
        profileIconId: json['profileIconId'],
        name: json['name'],
        puuid: json['puuid'],
        summonerLevel: json['summonerLevel'],
        revisionDate: json['revisionDate'],
        id: json['id'],
        accountId: json['accountId'],
        tier: json['tier'],
        rank: json['rank'],
        wins: json['wins'],
        losses: json['losses'],
        leaguePoints: json['leaguePoints']);
  }
}
